### DISCUSSION TITLE
Initial numbered release

### DISCUSSION THREAD

**Discussion by hsutter** [hsutter/cppfront#1025] [Announcements]
Initial feature set is complete, including documentation.

Modulo bugs of course, and with more new features still to be added as this project continues to evolve.

<hr /><em>This discussion was created from the release <a href='https://github.com/hsutter/cppfront/releases/tag/v0.7.0'>Initial numbered release</a>.</em>

**Comment by DyXel** [comment]
Just noticed you called it a _project_ now rather than a _experiment_. Is that intentional? 😛 

Congrats to everyone who has contributed! 🎉

**Comment by hsutter** [comment]
Yes, thanks very much again to everyone! For my talk at ACCU next month, I plan to update the "thank you" slide once again with every name of everyone who ever opened an issue or PR, and that font size is getting really really small to fit them all on a screen and I'm going to have to resort to a scrolling credits list I think. 😁 

Here's the CppCon 2023 slide for comparison:

![image](https://github.com/hsutter/cppfront/assets/1801526/594a1f05-dfeb-4486-8b17-5ef526de623d)

Since the above, there are ~300 new issues/PRs including from many new people. When I update this for each talk, I manually go through every issue and PR since the last talk to try to do my best not to miss anyone, and I'll do that again this time. (Disclaimer: As I've said before, at some point this isn't going to scale and I'll stop being able to keep up properly, but I'll keep it up as long as I reasonably can.)

Again, thanks everyone!

**Reply by DyXel** [reply]
Sweet! Let's keep working to make Cpp2 and C++ the best it can be 💪🏻

**Reply by mike919192** [reply]
I'm late to the announcement, but still congrats!  I haven't had much time lately to experiment with cppfront but I hope to again soon!

**Comment by hsutter** [comment]
> For my talk at ACCU next month, I plan to update the "thank you" slide once again with every name of everyone who ever opened an issue or PR

Why wait... I decided to update it now and went through the ~300 new issues/PRs to find all the new names. Here is the update as of today... the dotted line is the actual slide bounds for comparison with the above, so I'll need to shrink the font or move to a scrolling list.

![image](https://github.com/hsutter/cppfront/assets/1801526/525cf1d3-8017-44b8-89a4-eadb4c6f5e8d)

**Reply by gregmarr** [reply]
You could gain some space by moving the repo path up to the "Thank you!" line, and combining the two sub-lines.  Then maybe adjusting the bounds of the columns will cause some names to not wrap.  Also, I thought at first that the `> 1,000` was just a bullet.  `1,000+` would be clearer. (postfix unary operators for the win!)

> Thank you to github.com/hsutter/**cppfront** participants!
> 1,000+ issues and PRs, 160+ contributors

**Comment by AbhinavK00** [comment]
So now that the initial set of features is complete, whats next? Exception Handling? Lifetime checker?
Or something regarding package managers?

**Reply by hsutter** [reply]
In the next couple of months: Catching up on issues/PRs, and metaprogramming.

I need to review @MaxSagebaum's regex PR, @filipsajdak's is/as PR (Filip please let me know as soon as it's ready to review and I'll prioritize it, you've waited a long time), and all the other PRs.

I need to look at open issues starting with high-priority bugs.

You can all help please: I've opened a **new issue #1026** to ask for help identifying the high-priority issues and PRs.

All of that has been piling up while I spent the last few months mostly writing documentation (as you all rightly requested! I'm glad I did it). I did a few issue and PRs here and there, but writing and reviewing docs took time away from writing and reviewing code so now it's time to work on reducing the debt there.

**Comment by MaxSagebaum** [comment]
At least for the commits you could use: ` git log --format='%an' | sort -u` This should give you all names quite simply.

**Reply by jaakristioja** [reply]
Or something like `git shortlog -sne` if you want to see the commit counts as well (to eliminate duplicates, use a [.mailmap file](https://git-scm.com/docs/gitmailmap)).
